### PR TITLE
Simulation tests for smart contracts

### PR DESCRIPTION
**Pull request by fkneeland-figure**
## Description

In this pr I added a provwasm wrapper that added a smart contract in genesis and then had weighted operations to store, instantiate and execute a simple smart contract.

closes: #392 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by arnabmitra** [comment]
lgtm to me ..i see some linter error on unchanged files..so may just require the change i out in my PR for github actions.

**Review by arnabmitra — DISMISSED** [review]

File: app/app.go [hunk 1]
```diff
@@ -715,6 +715,7 @@ func New(
 		name.NewAppModule(appCodec, app.NameKeeper, app.AccountKeeper, app.BankKeeper),
 		attribute.NewAppModule(appCodec, app.AttributeKeeper, app.AccountKeeper, app.BankKeeper, app.NameKeeper),
 		msgfeesmodule.NewAppModule(appCodec, app.MsgFeesKeeper, app.interfaceRegistry),
+		//provwasm.NewWrapper(appCodec, &app.WasmKeeper, app.StakingKeeper, app.AccountKeeper, app.BankKeeper, app.NameKeeper),
```
Comment (arnabmitra) [inline app/app.go:718]: curious,, is this required?

Comment (fkneeland-figure) [reply app/app.go:718]: I ran into an issue with the import-export tests and was testing some things out... unfortunately got to wait on this pr 😢

**Comment by fkneeland-figure** [comment]
All the lints errors are unrelated and should be resolved with @arnabmitra pr.  

This should **not** be ported to main as the smart contract requires provenance v1.8.0 and so it won't work on current main.

Should be ready for review now. 🤞

**Review by dwedul-figure — APPROVED** [review]
LGTM (as long as the checks all pass).

**Review by arnabmitra — APPROVED** [review]

**Review by iramiller — APPROVED** [review]
Looks good ... is also a little bit of overlap with #94

**Pull request merged**